bigquery/google/cloud/bigquery/job.py

```diff
@@ -1880,11 +1881,25 @@ def done(self, retry=DEFAULT_RETRY):
         :rtype: bool
         :returns: True if the job is complete, False otherwise.
         """
+        # Since the API to getQueryResults can hang up to the timeout value
+        # (default of 10 seconds), set the timeout parameter to ensure that
+        # the timeout from the futures API is respected. See:
+        # https://github.com/GoogleCloudPlatform/google-cloud-python/issues/4135
+        timeout_ms = None
+        if self._done_timeout is not None:
+            # Subtract a buffer of a 100 milliseconds for context switching,
+            # network latency, etc.
+            timeout_ms = int(1000 * self._done_timeout) - 100
+            timeout_ms = max(min(timeout_ms, 10000), 0)
+            self._done_timeout -= (timeout_ms / 1000.0)
+            self._done_timeout = max(0, self._done_timeout)
+
         # Do not refresh is the state is already done, as the job will not
         # change once complete.
         if self.state != _DONE_STATE:
             self._query_results = self._client._get_query_results(
-                self.job_id, retry)
+                self.job_id, retry,
```

Comment:
I don't see anything here that deals with the inevitable timeouterror. Also, what if the timeouterror isn't due to the job not being done but is instead due to network issues?

Reply:
Sorry, I'm not sure to do if I get a timeout error. Return False?

Does the `reload()` call have the same problem (also used in the shared superclass)?

Reply:
For future reference, we discussed in person. The timeout variable in this instance is being used as a parameter to the API request, not in the sense that the futures API uses timeout.